packages/x-components/tests/unit/main-scroll.spec.ts

```diff
@@ -166,6 +169,44 @@ describe('testing MainScroll component', () => {
         });
       });
 
+      it('restores the scroll with transitions enabled', () => {
+        const template = `
+        <MainScroll v-bind="{ threshold, margin, useWindow }">
+          <div data-test="main-scroll">
+            <FadeAndSlide>
+              <MainScrollItem
+                v-for="item in items"
+                class="item"
+                tag="article"
+                :item="item"
+                :data-test="item.id">
+                {{ item.id }}
+              </MainScrollItem>
+            </FadeAndSlide>
+          </div>
+        </MainScroll>
+      `;
+        const { restoreScrollToItem, getItem } = renderMainScroll({
+          ...defaultParameters,
+          template
+        });
+
+        cy.getByDataTest('main-scroll').should('have.class', 'x-main-scroll--no-transition');
+        restoreScrollToItem(5);
+        cy.getByDataTest('main-scroll').should('not.have.class', 'x-main-scroll--no-transition');
+
+        cy.log('Item 5 should be the first one.');
+        getItem(5).should($item5 => {
+          expect($item5.get(0).getBoundingClientRect().top).to.be.eq(0);
+        });
+        cy.log("Item 4 shouldn't be visible");
+        getItem(4).should($item4 => {
+          const item4Bounds = $item4.get(0).getBoundingClientRect();
+          const item4Bottom = item4Bounds.top + item4Bounds.height;
+          expect(item4Bottom).to.be.eq(0);
+        });
```

Comment:
There is an issue with this part of the test, and it is that it still works without the new CSS you added disabling the transitions. The reason appears to be that Cypress executes it immediately, before the transition is run.

I think you can fix it by just replacing the `getItem(5)` code with this one which will wait until it is visible (opacity: 1) before checking its position.

```ts
getItem(5)
            .should('be.visible')
            .should($item5 => {
              expect($item5.get(0).getBoundingClientRect().top).to.be.eq(0);
            });
```

If you do this, then the `x-main-scroll--no-transition` class checks are unnecessary, as you are testing implementation details, how the scroll is restored when there are transitions instead of that the scroll is properly restored like you are doing below. I would be fine with testing the class if there was no other way to test it, but I'd rather rely on testing that the scroll is restored properly like you are already doing. Moreover these class checks are failing for me and I'm not able to make them work, while I see them passing in the CI, so I have no confidence at all 🤣. 

![image](https://user-images.githubusercontent.com/68222542/145964103-ad61db2c-ce08-4b24-9eef-8aff22910401.png)